cylc/flow/tests/test_rundb.py

```diff
@@ -170,5 +196,76 @@ def test_upgrade_hold_swap():
         assert not dao.upgrade_is_held()
 
 
+def test_upgrade_to_platforms(set_up_globalrc):
+    """Test upgrader logic for platforms in the database.
+    """
+    # Set up the globalrc
+    set_up_globalrc(GLOBALRC)
+
+    # task name, cycle, user_at_host, batch_system
+    initial_data = [
+        ('hpc_with_pbs', '1', 'hpcl1', 'pbs'),
+        ('desktop_with_bg', '1', 'desktop01', 'background'),
+        ('slurm_no_host', '1', '', 'slurm'),
+        ('hpc_bg', '1', 'hpcl1', 'background'),
+        ('username_given', '1', 'slartibartfast@hpcl1', 'pbs')
```

Comment:
```suggestion
        ('username_given', '1', 'slartibartfast@deepthought', 'pbs')
```

Reply:
slartibartfast@the_earth, surely @oliver-sanders ?

Reply:
slartibartfast@campain-for-real-time.mg